homeassistant/components/bosch_shc/alarm_control_panel.py

```diff
@@ -0,0 +1,95 @@
+"""Platform for alarm control panel integration."""
+from __future__ import annotations
+
+from boschshcpy import SHCIntrusionSystem, SHCSession
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMING,
+    STATE_ALARM_DISARMED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DATA_SESSION, DOMAIN
+from .entity import SHCDomainEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the alarm control panel platform."""
+    session: SHCSession = hass.data[DOMAIN][config_entry.entry_id][DATA_SESSION]
+
+    alarm_control_panel = IntrusionSystemAlarmControlPanel(
+        domain=session.intrusion_system,
+        parent_id=session.information.unique_id,
+        entry_id=config_entry.entry_id,
+    )
+
+    async_add_entities([alarm_control_panel])
+
+
+class IntrusionSystemAlarmControlPanel(SHCDomainEntity, AlarmControlPanelEntity):
+    """Representation of SHC intrusion detection control."""
+
+    _attr_code_arm_required: bool = False
+    _attr_should_poll: bool = False
+    _attr_supported_features: AlarmControlPanelEntityFeature = (
+        AlarmControlPanelEntityFeature.ARM_AWAY
+        | AlarmControlPanelEntityFeature.ARM_HOME
+        | AlarmControlPanelEntityFeature.ARM_CUSTOM_BYPASS
+    )
+
+    @property
+    def state(self) -> StateType:
```

Comment:
It would be better to have a dict with `{SHCstate : HA_state}` and then use a dict lookup instead of all these if-statements